### PR TITLE
Introduce a che-editors file

### PR DESCRIPTION
**Pull request by ericwill**
Signed-off-by: Eric Williams <ericwill@redhat.com>

### What does this PR do?
Contributes a `che-editors.yaml` file, similar to the `che-theia-plugins.yaml` file, but for editors.

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
Closes eclipse/che#18214

### How to test this PR?
N/A for now

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: che-editors.yaml [hunk 1]
```diff
@@ -0,0 +1,192 @@
+version: 1.0.0
+editors:
+  - repository:
+      url: https://github.com/eclipse/che-theia
+      revision: master
+      endpoints:
+        - name: "theia"
```
Comment (benoitf) [inline che-editors.yaml:7]: how do we define that it'll be `next` and not `latest` ?

Comment (benoitf) [reply che-editors.yaml:7]: also I don't find the `latest` definition

Comment (ericwill) [reply che-editors.yaml:7]: We could use the `id` system, similar to for plugins?

Also about the repository fields, do we need revision? Some repos don't have anything interesting in them, or don't have tags create to match the release.

Comment (benoitf) [reply che-editors.yaml:7]: Ok for `id` and let's remove repository for now if we don't see how to use it for now then.

File: che-editors.yaml [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+          discoverable: false
+      - name: "theia-redirect-3"
+        public: true
+        targetPort: 13133
+        attributes:
+          protocol: http
+          discoverable: false
+    containers:
+      - name: theia-ide
+        image: "quay.io/eclipse/che-theia:7.22.2"
```
Comment (benoitf) [inline che-editors.yaml:146]:
```suggestion
        image: "quay.io/eclipse/che-theia:7.23.0"
```

File: che-editors.yaml [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+        volumes:
+          - name: remote-endpoint
+            path: "/remote-endpoint"
+            ephemeral: true
+        env:
+          - name: PLUGIN_REMOTE_ENDPOINT_EXECUTABLE
+            value: /remote-endpoint/plugin-remote-endpoint
+          - name: REMOTE_ENDPOINT_VOLUME_NAME
+            value: remote-endpoint
+  - id: eclipse/che-theia/7.22.2
```
Comment (benoitf) [inline che-editors.yaml:88]: let's make it to 7.23
```suggestion
  - id: eclipse/che-theia/7.23.0
```

File: che-editors.yaml [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+          - name: THEIA_HOST
+            value: 127.0.0.1
+        volumes:
+          - name: plugins
+            path: "/plugins"
+        mountSources: true
+        memoryLimit: "512M"
+    initContainers:
+      - name: remote-runtime-injector
+        image: "quay.io/eclipse/che-theia-endpoint-runtime-binary:7.22.2"
```
Comment (benoitf) [inline che-editors.yaml:163]:
```suggestion
        image: "quay.io/eclipse/che-theia-endpoint-runtime-binary:7.23.0"
```

**Review by benoitf — APPROVED** [review]
over night 7.23.0 was tagged.
I'm not able yet to test as there is no client but let's update to 7.23.0 and merge, we'll be able to fix something later if there is something missing

**Pull request merged**